File: DoitMission_08/Doit08_Java/app/src/main/java/com/example/doit08_java/MenuActivity.java
package com.example.doit08_java;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MenuActivity extends AppCompatActivity {

    public static final int MENU_REQUEST_CODE = 201;
    public static final String MENU_REQUEST = "data";

    private Button custom;
    private Button sales;
    private Button product;
    private Intent intent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_menu);

        custom = findViewById(R.id.custom);
        sales = findViewById(R.id.sales);
        product = findViewById(R.id.product);


        custom.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                intent = new Intent(getApplicationContext(),CustomActivity.class);
                intent.putExtra(MENU_REQUEST,custom.getText().toString());
                startActivityForResult(intent,MENU_REQUEST_CODE);
            }
        });

        sales.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                intent = new Intent(getApplicationContext(),SalesActivity.class);
                intent.putExtra(MENU_REQUEST,sales.getText().toString());
                startActivityForResult(intent,MENU_REQUEST_CODE);
            }
        });

        product.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                intent = new Intent(getApplicationContext(),ProductActivity.class);
                intent.putExtra(MENU_REQUEST,product.getText().toString());
                startActivityForResult(intent,MENU_REQUEST_CODE);
            }
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if(requestCode == MENU_REQUEST_CODE){
            if(resultCode == RESULT_OK){
                Toast.makeText(getApplicationContext(),data.getStringExtra(MENU_REQUEST)+ " 액티비티에서 넘어옴",Toast.LENGTH_SHORT).show();
            }
        }
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        finish();
    }
}

File: DoitMission_08/Doit08_Kotlin/app/src/main/java/com/example/doit08_kotlin/MenuActivity.kt
package com.example.doit08_kotlin

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_menu.*
import org.jetbrains.anko.startActivity
import org.jetbrains.anko.toast

class MenuActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_menu)

        custom.setOnClickListener {
            val intent = Intent(this,CustomActivity::class.java)
            intent.putExtra("data",custom.text)
            startActivityForResult(intent,301)
        }

        product.setOnClickListener {
            val intent = Intent(this,ProductActivity::class.java)
            intent.putExtra("data",product.text)
            startActivityForResult(intent,301)
        }

        sales.setOnClickListener {
            val intent = Intent(this,SalesActivity::class.java)
            intent.putExtra("data",sales.text)
            startActivityForResult(intent,301)
        }
    }

    override fun onBackPressed() {
        super.onBackPressed()
        setResult(0,intent)

    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if(resultCode == Activity.RESULT_OK){
            val dataA = data!!.getStringExtra("data")
            toast(dataA + "에서 넘어옴")
        }
    }
}
